### PR TITLE
SignalR additional xml docs

### PR DESCRIPTION
**Pull request by analogrelay**
Reviewed our xml docs as both ramping back up after leave and docathon. Things look pretty good, added a few things to some public methods.

### CODE REVIEW

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 1]
```diff
@@ -63,6 +63,31 @@ public partial class HubConnection
         private ConnectionState _connectionState;
         private int _serverProtocolMinorVersion;
 
+        /// <summary>
+        /// Occurs when the connection is closed, either by a call to <see cref="StopAsync(CancellationToken)"/> or a connection error.
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: or the server closed the connection gracefully

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 2]
```diff
@@ -63,6 +63,31 @@ public partial class HubConnection
...
+        /// If this event was triggered from a connection error, the <see cref="Exception"/> that occurred will be passed in as the
+        /// sole argument to this handler. If this event was triggered from a call to <see cref="StopAsync(CancellationToken)"/>, then
+        /// the argument will be <see langword="null"/>.
+        /// </remarks>
+        /// <example>
+        /// The following example attaches a handler to the <see cref="Closed"/> event, and checks the provided argument to determine
+        /// if there was an error:
+        ///
+        /// <code>
+        /// connection.Closed += (exception) => {
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: nit: `{` on newline

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 3]
```diff
@@ -985,11 +1010,17 @@ private async Task ReceiveLoop(ConnectionState connectionState)
             }
         }
 
+        /// <summary>
+        /// This method is for internal framework use and should not be called by user code.
```
Comment (BrennanConroy) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Or just make it private :)

File: src/SignalR/common/Http.Connections/src/HttpConnectionContextExtensions.cs [hunk 4]
```diff
@@ -9,6 +9,15 @@ namespace Microsoft.AspNetCore.Http.Connections
 {
     public static class HttpConnectionContextExtensions
     {
+        /// <summary>
+        /// Gets the <see cref="HttpContext"/> associated with the connection, if there is one.
+        /// </summary>
+        /// <param name="connection">The <see cref="ConnectionContext"/> representing the connection.</param>
+        /// <returns>The <see cref="HttpContext"/> associated with the connection, or <see langword="null"/> if the connection is not HTTP-based</returns>
```
Comment (BrennanConroy) [inline src/SignalR/common/Http.Connections/src/HttpConnectionContextExtensions.cs]: full stop

**Review by BrennanConroy — APPROVED** [review]

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 5]
```diff
@@ -63,6 +63,31 @@ public partial class HubConnection
         private ConnectionState _connectionState;
         private int _serverProtocolMinorVersion;
 
+        /// <summary>
+        /// Occurs when the connection is closed, either by a call to <see cref="StopAsync(CancellationToken)"/> or a connection error.
+        /// </summary>
+        /// <remarks>
+        /// If this event was triggered from a connection error, the <see cref="Exception"/> that occurred will be passed in as the
+        /// sole argument to this handler. If this event was triggered from a call to <see cref="StopAsync(CancellationToken)"/>, then
```
Comment (mikaelm12) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: You might want to add the server closing the connection gracefully case here as well

**Review by mikaelm12 — APPROVED** [review]

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs (hunk 1 above)
Comment (anurse) [reply src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: True, will update.

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs (hunk 3 above)
Comment (anurse) [reply src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: I didn't want to bog this PR down with that, but I can just remove the doc comment. It was public in 2.2 so our standard policy is to deprecate it over one release first. See https://github.com/aspnet/AspNetCore/issues/7977 for more. I'll remove these changes and we can just discuss things there.

**Review by halter73 — APPROVED** [review]

File: src/SignalR/common/Http.Connections/src/HttpConnectionContextExtensions.cs [hunk 6]
```diff
@@ -9,6 +9,15 @@ namespace Microsoft.AspNetCore.Http.Connections
...
     {
+        /// <summary>
+        /// Gets the <see cref="HttpContext"/> associated with the connection, if there is one.
+        /// </summary>
+        /// <param name="connection">The <see cref="ConnectionContext"/> representing the connection.</param>
+        /// <returns>The <see cref="HttpContext"/> associated with the connection, or <see langword="null"/> if the connection is not HTTP-based.</returns>
+        /// <remarks>
+        /// SignalR connections can run on top of HTTP transports like WebSockets or Long Polling, or other non-HTTP transports. As a result,
+        /// this method can sometimes return <see langword="null"/> depending on the configuration of your application.
+        /// </remarks>
```
Comment (halter73) [inline src/SignalR/common/Http.Connections/src/HttpConnectionContextExtensions.cs]: Beware that this won't show up in intellisense or the HttpConnectionContextExtensions generated docs. To see this you will have to got to the generated docs page dedicated exclusively to the GetHttpContext() method.

Comment (anurse) [reply src/SignalR/common/Http.Connections/src/HttpConnectionContextExtensions.cs]: Yeah, I think I'll put `or <see langword="null" /> if there is no HTTP context` in the summary (and leave this remark that adds more detail).

**Comment by anurse** [comment]
https://github.com/aspnet/AspNetCore/issues/7265 - which should now be skipped. Rerunning

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 7]
```diff
@@ -63,13 +63,40 @@ public partial class HubConnection
...
+        /// <code>
+        /// connection.Closed += (exception) =>
+        /// {
+        ///     if (exception == null)
+        ///     {
+        ///         Console.WriteLine("Connection closed without error.");
+        ///     }
+        ///     else
+        ///     {
+        ///         Console.WriteLine($"Connection closed due to an error: {exception.Message}");
```
Comment (halter73) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Nit: I normally prefer printing $"{exception}" over $"{exception.Message}" so you get the Exception type and stack trace too.

**Review by halter73 — APPROVED** [review]

**Pull request merged**